Remove RenderFullScreen and RenderTreeBuilderFullScreen

nt1m opened the pull request:
#### 5ca2229ca6786088cfef0154b43c850d26e8b4c9
<pre>
Remove RenderFullScreen and RenderTreeBuilderFullScreen
<a href="https://bugs.webkit.org/show_bug.cgi?id=246065">https://bugs.webkit.org/show_bug.cgi?id=246065</a>
rdar://100793009

Reviewed by Simon Fraser.

RenderFullscreen was added to prevent visible layout shifts while a certain element is being fullscreened. This is now unnecessary with the current CSS for fullscreen elements,
since fullscreen will expand and fixed position the element immeditately. It has no performance benefit given making an element fullscreen already schedules a full style rebuild and changes
layout for the whole ancestor chain for the current fullscreen element.

Remove it since it is not trivial to maintain, makes porting to top layer and future projects like LFC harder.

* LayoutTests/compositing/no-compositing-when-fulll-screen-is-present-expected.txt:
* LayoutTests/fullscreen/full-screen-layer-dump-expected.txt:
* LayoutTests/fullscreen/full-screen-placeholder-expected.txt: Removed.
* LayoutTests/fullscreen/full-screen-placeholder.html: Removed.
* LayoutTests/fullscreen/fullscreen-restore-scroll-position.html:
* LayoutTests/platform/glib/fullscreen/full-screen-placeholder-expected.txt: Removed.
* LayoutTests/platform/gtk/compositing/no-compositing-when-fulll-screen-is-present-expected.txt:
* Source/WebCore/Sources.txt:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::willEnterFullscreen):
(WebCore::FullscreenManager::didEnterFullscreen):
(WebCore::FullscreenManager::willExitFullscreen):
(WebCore::FullscreenManager::didExitFullscreen):
(WebCore::FullscreenManager::setFullscreenRenderer): Deleted.
(WebCore::FullscreenManager::fullscreenRenderer const): Deleted.
(WebCore::unwrapFullscreenRenderer): Deleted.
* Source/WebCore/dom/FullscreenManager.h:
* Source/WebCore/page/FrameView.cpp:
* Source/WebCore/rendering/RenderFullScreen.cpp: Removed.
* Source/WebCore/rendering/RenderFullScreen.h: Removed.
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::isDescendantOfFullScreenLayer):
* Source/WebCore/rendering/RenderVideo.cpp:
(WebCore::placeholder): Deleted.
(WebCore::RenderVideo::offsetLeft const): Deleted.
(WebCore::RenderVideo::offsetTop const): Deleted.
(WebCore::RenderVideo::offsetWidth const): Deleted.
(WebCore::RenderVideo::offsetHeight const): Deleted.
* Source/WebCore/rendering/RenderVideo.h:
* Source/WebCore/rendering/updating/RenderTreeBuilder.cpp:
(WebCore::RenderTreeBuilder::RenderTreeBuilder):
(WebCore::RenderTreeBuilder::destroy):
(WebCore::RenderTreeBuilder::createPlaceholderForFullScreen): Deleted.
* Source/WebCore/rendering/updating/RenderTreeBuilder.h:
(WebCore::RenderTreeBuilder::continuationBuilder):
(WebCore::RenderTreeBuilder::fullScreenBuilder): Deleted.
* Source/WebCore/rendering/updating/RenderTreeBuilderBlock.cpp:
(WebCore::RenderTreeBuilder::Block::attachIgnoringContinuation):
* Source/WebCore/rendering/updating/RenderTreeBuilderFullScreen.cpp: Removed.
* Source/WebCore/rendering/updating/RenderTreeBuilderFullScreen.h: Removed.
* Source/WebCore/rendering/updating/RenderTreeBuilderInline.cpp:
(WebCore::RenderTreeBuilder::Inline::splitInlines):
* Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
(WebCore::RenderTreeUpdater::createRenderer):

Canonical link: <a href="https://commits.webkit.org/255641@main">https://commits.webkit.org/255641@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b8c9416ebe098cc54f1df18acc42cb4b09d35199

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93168 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/2369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/23819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/102874 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/163167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/97171 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/2372 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/30692 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/85558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/98991 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98837 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/2372 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/79639 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/85558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/2372 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/23819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/85558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/66/builds/37082 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/23819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/34898 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/23819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/63/builds/38769 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/79639 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1796 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/65/builds/40698 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/23819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -2632,13 +2631,14 @@ enum class FullScreenDescendant { Yes, No, NotApplicable };
-        ASSERT_NOT_REACHED();
+    auto* fullScreenRenderer = dynamicDowncast<RenderLayerModelObject>(fullScreenElement->renderer());
+    auto* fullScreenLayer = fullScreenRenderer->layer();
+    if (!fullScreenRenderer || !fullScreenLayer) {
```
You can remove the parens.

nt1m committed:
Remove RenderFullScreen and RenderTreeBuilderFullScreen
https://bugs.webkit.org/show_bug.cgi?id=246065
rdar://100793009

Reviewed by Simon Fraser.

RenderFullscreen was added to prevent visible layout shifts while a certain element is being fullscreened. This is now unnecessary with the current CSS for fullscreen elements,
since fullscreen will expand and fixed position the element immeditately. It has no performance benefit given making an element fullscreen already schedules a full style rebuild and changes
layout for the whole ancestor chain for the current fullscreen element.

Remove it since it is not trivial to maintain, makes porting to top layer and future projects like LFC harder.

* LayoutTests/compositing/no-compositing-when-fulll-screen-is-present-expected.txt:
* LayoutTests/fullscreen/full-screen-layer-dump-expected.txt:
* LayoutTests/fullscreen/full-screen-placeholder-expected.txt: Removed.
* LayoutTests/fullscreen/full-screen-placeholder.html: Removed.
* LayoutTests/fullscreen/fullscreen-restore-scroll-position.html:
* LayoutTests/platform/glib/fullscreen/full-screen-placeholder-expected.txt: Removed.
* LayoutTests/platform/gtk/compositing/no-compositing-when-fulll-screen-is-present-expected.txt:
* Source/WebCore/Sources.txt:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::willEnterFullscreen):
(WebCore::FullscreenManager::didEnterFullscreen):
(WebCore::FullscreenManager::willExitFullscreen):
(WebCore::FullscreenManager::didExitFullscreen):
(WebCore::FullscreenManager::setFullscreenRenderer): Deleted.
(WebCore::FullscreenManager::fullscreenRenderer const): Deleted.
(WebCore::unwrapFullscreenRenderer): Deleted.
* Source/WebCore/dom/FullscreenManager.h:
* Source/WebCore/page/FrameView.cpp:
* Source/WebCore/rendering/RenderFullScreen.cpp: Removed.
* Source/WebCore/rendering/RenderFullScreen.h: Removed.
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::isDescendantOfFullScreenLayer):
* Source/WebCore/rendering/RenderVideo.cpp:
(WebCore::placeholder): Deleted.
(WebCore::RenderVideo::offsetLeft const): Deleted.
(WebCore::RenderVideo::offsetTop const): Deleted.
(WebCore::RenderVideo::offsetWidth const): Deleted.
(WebCore::RenderVideo::offsetHeight const): Deleted.
* Source/WebCore/rendering/RenderVideo.h:
* Source/WebCore/rendering/updating/RenderTreeBuilder.cpp:
(WebCore::RenderTreeBuilder::RenderTreeBuilder):
(WebCore::RenderTreeBuilder::destroy):
(WebCore::RenderTreeBuilder::createPlaceholderForFullScreen): Deleted.
* Source/WebCore/rendering/updating/RenderTreeBuilder.h:
(WebCore::RenderTreeBuilder::continuationBuilder):
(WebCore::RenderTreeBuilder::fullScreenBuilder): Deleted.
* Source/WebCore/rendering/updating/RenderTreeBuilderBlock.cpp:
(WebCore::RenderTreeBuilder::Block::attachIgnoringContinuation):
* Source/WebCore/rendering/updating/RenderTreeBuilderFullScreen.cpp: Removed.
* Source/WebCore/rendering/updating/RenderTreeBuilderFullScreen.h: Removed.
* Source/WebCore/rendering/updating/RenderTreeBuilderInline.cpp:
(WebCore::RenderTreeBuilder::Inline::splitInlines):
* Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
(WebCore::RenderTreeUpdater::createRenderer):

Canonical link: https://commits.webkit.org/255641@main
